core/src/main/java/hudson/Util.java

```diff
@@ -1182,16 +1242,21 @@ public static FileSet createFileSet(@Nonnull File baseDir, @Nonnull String inclu
      */
     public static void createSymlink(@Nonnull File baseDir, @Nonnull String targetPath,
             @Nonnull String symlinkPath, @Nonnull TaskListener listener) throws InterruptedException {
+        File fileForSymlink = new File(baseDir, symlinkPath);
         try {
-            Path path = fileToPath(new File(baseDir, symlinkPath));
+            Path pathForSymlink = fileToPath(fileForSymlink);
             Path target = Paths.get(targetPath, MemoryReductionUtil.EMPTY_STRING_ARRAY);
 
+            if (createSymlinkAtomic(pathForSymlink, fileForSymlink, target, symlinkPath)) {
```

Comment:
seems like REPLACE_EXISTING should be used before trying to delete the file to minimize the race condition further?